recipes/install.rb

```diff
@@ -78,4 +66,65 @@
   action :create
 end
 
+directory node['data']['dir'] do
+  owner 'root'
+  group 'root'
+  mode '0775'
+  action :create
+  not_if { ::File.directory?(node['data']['dir']) }
+end
 
+directory node['kkafka']['data_volume']['logs_dir'] do
+  owner node['kkafka']['user']
+  group node['kkafka']['group']
+  mode '770'
+  action :create
+  recursive true
+  not_if { File.directory?(node['kkafka']['data_volume']['logs_dir']) }
+end
+
+kafka_log_dirs.each do |dir|
+  bash "Move Kafka logs #{dir} to data volume" do
+    user 'root'
+    code <<-EOH
+      set -e
+      mv -f #{dir}/* #{node['kkafka']['data_volume']['logs_dir']}
+      rm -rf #{dir}
+    EOH
+    only_if { conda_helpers.is_upgrade }
+    only_if { File.directory?(dir)}
+    not_if { File.symlink?(dir)}
+  end
+
+  link dir do
+    owner node['kkafka']['user']
+    group node['kkafka']['group']
+    mode '770'
+    to node['kkafka']['data_volume']['logs_dir']
+  end
+end
+
+directory node['kkafka']['data_volume']['app_log_dir'] do
+  owner node['kkafka']['user']
+  group node['kkafka']['group']
+  mode '0750'
+end
+
+bash 'Move Kafka application logs to data volume' do
+  user 'root'
+  code <<-EOH
+    set -e
+    mv -f #{node['kkafka']['log_dir']}/* #{node['kkafka']['data_volume']['app_log_dir']}
+    rm -rf #{node['kkafka']['log_dir']}
```

Comment:
Would be great to have the `_deprecated` move here as well. Just in case we need to recover stuff.